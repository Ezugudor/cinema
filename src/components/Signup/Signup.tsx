import React, { useState } from "react";
import { Movie } from "../../models/Movie";
import { Theme } from "../../models/Theme";
import styles from "./Signup.module.css";

export interface SignupProps {
    theme: Theme;
}

//TODO: Move to LocalStorage
const tempMovies: Movie[] = [
    { title: "Fatherhood", time: "18:00:00" },
    { title: "Spiderman", time: "18:00:00" },
    { title: "Lacasa De Papel 5", time: "20:00:00" },
]

export const Signup: React.FunctionComponent<SignupProps> = props => {

    const [firstname, setFirstname] = useState('');
    const [lastname, setLastname] = useState('');


    return (
        <div className={styles.FormWrapper}>
            {firstname}
            <form action="#">
                <input type="text" name="firstname" value={firstname} onChange={e => { setFirstname(e.target.value) }} />
                <input type="text" name="lastname" value={lastname} onChange={e => { setLastname(e.target.value) }} />
                <select name="" id="">
                    {
                        tempMovies.map((movie: Movie, index) => {
                            return <option value={movie.title} key={index}>{movie.title} - {movie.time}</option>
                        })
                    }
                </select>
            </form>
        </div>
    );
};
