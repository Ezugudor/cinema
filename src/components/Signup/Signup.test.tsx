import React from "react";
import Adapter from "enzyme-adapter-react-16";
import { Signup } from "./Signup";
import { configure, shallow } from "enzyme";


configure({ adapter: new Adapter() });

describe("Signup Component", () => {
    const appMock = {};

    it("should render the Signup Component", () => {
        const wrapper = shallow(
            <Signup
                theme={{ background: '#fff', textColor: "black" }}
            />
        );
        expect(wrapper.html()).toMatchSnapshot();
    });

});
