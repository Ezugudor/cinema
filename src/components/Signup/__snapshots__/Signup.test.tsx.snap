// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup Component should render the Signup Component 1`] = `"<div class=\\"FormWrapper\\"><form action=\\"#\\"><input type=\\"text\\" name=\\"firstname\\" value=\\"\\"/><input type=\\"text\\" name=\\"lastname\\" value=\\"\\"/><select name=\\"\\" id=\\"\\"><option value=\\"Fatherhood\\">Fatherhood - 18:00:00</option><option value=\\"Spiderman\\">Spiderman - 18:00:00</option><option value=\\"Lacasa De Papel 5\\">Lacasa De Papel 5 - 20:00:00</option></select></form></div>"`;
